panther_hardware_interfaces/test/unit/panther_system/robot_driver/test_panther_robot_driver.cpp

```diff
@@ -0,0 +1,216 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <stdexcept>
+#include <vector>
+
+#include <gmock/gmock.h>
+
+#include <panther_hardware_interfaces/panther_system/robot_driver/canopen_manager.hpp>
+#include <panther_hardware_interfaces/panther_system/robot_driver/driver.hpp>
+#include <panther_hardware_interfaces/panther_system/robot_driver/panther_robot_driver.hpp>
+#include <panther_hardware_interfaces/panther_system/robot_driver/robot_driver.hpp>
+
+#include "utils/fake_can_socket.hpp"
+#include "utils/mock_driver.hpp"
+#include "utils/test_constants.hpp"
+
+#include "panther_utils/test/test_utils.hpp"
+
+class PantherRobotDriverWrapper : public panther_hardware_interfaces::PantherRobotDriver
+{
+public:
+  PantherRobotDriverWrapper(
+    const panther_hardware_interfaces::CANopenSettings & canopen_settings,
+    const panther_hardware_interfaces::DrivetrainSettings & drivetrain_settings,
+    const std::chrono::milliseconds activate_wait_time = std::chrono::milliseconds(1000))
+  : PantherRobotDriver(canopen_settings, drivetrain_settings, activate_wait_time)
+  {
+    mock_fl_motor_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockMotorDriver>>();
+    mock_fr_motor_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockMotorDriver>>();
+    mock_rl_motor_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockMotorDriver>>();
+    mock_rr_motor_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockMotorDriver>>();
+
+    mock_front_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockDriver>>();
+    mock_front_driver_->AddMotorDriver(
+      panther_hardware_interfaces::MotorNames::LEFT, mock_fl_motor_driver_);
+    mock_front_driver_->AddMotorDriver(
+      panther_hardware_interfaces::MotorNames::RIGHT, mock_fr_motor_driver_);
+
+    mock_rear_driver_ =
+      std::make_shared<::testing::NiceMock<panther_hardware_interfaces_test::MockDriver>>();
+    mock_rear_driver_->AddMotorDriver(
+      panther_hardware_interfaces::MotorNames::LEFT, mock_rl_motor_driver_);
+    mock_rear_driver_->AddMotorDriver(
+      panther_hardware_interfaces::MotorNames::RIGHT, mock_rr_motor_driver_);
+  }
+
+  void DefineDrivers() override
+  {
+    drivers_.emplace(panther_hardware_interfaces::DriverNames::FRONT, mock_front_driver_);
+    drivers_.emplace(panther_hardware_interfaces::DriverNames::REAR, mock_rear_driver_);
+  }
+
+  std::shared_ptr<::testing::NiceMock<panther_hardware_interfaces_test::MockDriver>>
+  GetMockFrontDriver()
+  {
+    return mock_front_driver_;
+  }
```

Comment:
Same as in `test_roboteq_robot_driver.cpp`, maybe better to make these attributes public? Applies as well to Lynx tests.

Reply:
@KmakD Are u going to address this as well?